service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.entity.Tag;
+import com.epam.esm.entity.User;
+import com.epam.esm.exception.ValidationException;
+import com.epam.esm.repository.UserRepository;
+import com.epam.esm.service.UserService;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+
+    public UserServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public User create(User entity) {
```

Comment:
Seems to me, you added unnecessary functionalities to User Service. Only get user would be enough.
<img width="553" alt="image" src="https://github.com/AkobirToshtemirov/GiftCertificates/assets/80680360/e3a19360-838f-4cc6-acb0-e73f5512caa0">